amethyst_renderer/src/system.rs

```diff
@@ -68,6 +71,7 @@ where
             pipe,
             renderer,
             cached_size,
+            event_vec: Vec::new(),
```

Comment:
Since you know you are going to receive events, you could pre-initialize it with a decent size. e.g 50? That would reduce load of the first few frames when the buffer gets resized.

Reply:
I went with a more conservative 20 events, but that was a good note.